test/e2e/federate.go

```diff
@@ -0,0 +1,184 @@
+/*
```

Comment:
I think a single golden-path test of kubefed2 should suffice. It shouldn't be necessary to test every available fixture.  The semantics of crud operations involve the potential for controller interaction, but creating a federated resource from a non-federated resource does not, so there should be no need to validate for all types.

Consider writing a unit test for the composition of the federated resource from the non-federated resource.  It should be possible to isolate the composition in a function that takes the type config and non-federated resource and does not require any api calls.

Reply:
The reason I did update an e2e was that I thought it to be valuable to be able to fetch a `real`  kubernetes resource from an actual API server, rather then composing one in a test function (which the test is doing anyways), and then create a federated resource against an actual API server. This ensures nuances that the API logic might include, viz. mismatched `defaulting`, `validation` across the k8s API and federation API are not overlooked.
The same reason applies to check this for all available types.
Let me know your thoughts.

Reply:
The federate resource test is included with `crud` now!